app/src/sharedTest/java/org/oppia/android/app/home/PromotedStoryListViewModelTest.kt

```diff
@@ -0,0 +1,525 @@
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [PromotedStoryListViewModel] data. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = PromotedStoryListViewModelTest.TestApplication::class,
+  manifest = Config.NONE
+)
+class PromotedStoryListViewModelTest {
+  @Inject
+  lateinit var context: Context
+
+  private val promotedStory1 = PromotedStory.newBuilder()
+    .setStoryId("id_1")
+    .setStoryName("Story 1")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+  private val promotedStory2 = PromotedStory.newBuilder()
+    .setStoryId("id_2")
+    .setStoryName("Story 2")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+  private val promotedStory3 = PromotedStory.newBuilder()
+    .setStoryId("id_3")
+    .setStoryName("Story 3")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testPromotedStoryListViewModelEquals_reflexiveStoryListOf2_isTrue() {
+    ActivityScenario.launch<HomeFragmentTestActivity>(
+      HomeFragmentTestActivity.createHomeFragmentTestActivity(context)
+    ).use {
+      it.onActivity {
+        val promotedStoryListViewModel = PromotedStoryListViewModel(
+          it,
+          listOf(
+            PromotedStoryViewModel(
+              /* activity = */ it,
+              /* internalProfileId = */1,
+              /* totalStoryCount = */3,
+              /* entytType = */"",
+              promotedStory1
```

Comment:
Nit: space after ``/`` (something to keep an eye out for to ensure formatting consistency). That being said, perhaps you can simplify the tests & reduce them a bit by doing something like:

```kotlin
private val createPromotedStoryViewModelList(activity: Activity, promotedStoryList: List<PromotedStory>): List<PromotedStoryViewModel> {
  return promotedStoryList.map {
    PromotedStoryViewModel(activity, internalProfileId = 1, totalStoryCount = promotedStoryList.size, entityType = "", promotedStory = it)
  }
}

private val createPromotedStoryListViewModel(activity: Activity, promotedStoryList: List<PromotedStory>): PromotedStoryListViewModel {
  return PromotedStoryListViewModel(activity, createPromotedStoryViewModelList(promotedStoryList))
}
```

(notice using named arguments instead of comments--these are clearer when they can be used & they are enforced by the compiler unlike comments).

Ditto elsewhere in this & the other tests. Let's use helpers to bring down the size of the tests by factoring out common boilerplate.

Reply:
Done